win32/tests/reals_win32/real_interlocked.h

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#ifndef REAL_INTERLOCKED_H
+#define REAL_INTERLOCKED_H
+
+#ifdef __cplusplus
+#include <cstdint>
+#include <cstddef>
+#else
+#include <stdint.h>
+#include <stddef.h>
+#endif
+
+#include "azure_macro_utils/macro_utils.h"
+
+#define R2(X) REGISTER_GLOBAL_MOCK_HOOK(X, real_##X);
+
+#ifdef _WIN64 
+#define REGISTER_INTERLOCKED_GLOBAL_MOCK_HOOK()     \
+    MU_FOR_EACH_1(R2,                               \

```

Comment:
makes me wonder if we could have something simpler with less repeated lines:

define REGISTER_BASE (has everything except 128)

if win64
define REGISTER_INTERLOCKED... REGISTER_BASE and 128
else
define REGISTER_INTERLOCKED... REGISTER_BASE 
endif

Reply:
Yup, clearly works :) Thanks!

---
In reply to: [460144666](https://github.com/Azure/azure-c-pal/pull/16#discussion_r460144666) [](ancestors = 460144666)